src/hydrogen/blas/gpu/Transpose.cu

```diff
@@ -55,23 +55,28 @@ __global__ void transpose_kernel(
             B[idx_out + ii*ldb] = tile[threadIdx.x][threadIdx.y+ii];
         }
     }
-    else
+    else if (do_anything)
```

Comment:
I think we need an `else` case to make sure all threads in the warp enter `cg::sync` at some point to prevent threads from hanging there.

Reply:
I made a mistake. That `if (do_anything)` shouldn't be there. I had that to see if it was sufficient; upon realizing it was incorrect, I thought I deleted it. Will update. Good catch and thanks!

Reply:
Fixed.